api_app/views.py

```diff
@@ -478,22 +479,238 @@ def aggregate_md5(self, request):
+        if organization:
+            return self.__aggregation_response_dynamic(
+                "file_mimetype", users=users_of_organization
+            )
+        else:
+            return self.__aggregation_response_dynamic(
+                "file_mimetype", user=current_user
+            )
+
+    @action(
+        url_path="aggregate/org/observable_name",
+        detail=False,
+        methods=["GET"],
+    )
+    @cache_action_response(timeout=60 * 5)
+    def aggregate_org_observable_name(self, request):
+        current_user = request.user
+        organization = None
+        users_of_organization = None
+        try:
+            # Check if the user has a membership
+            if current_user.has_membership():
+                organization = current_user.membership.organization
+                if organization:
+                    users_of_organization = [
+                        membership.user for membership in organization.members.all()
+                    ]
+
+        except AttributeError:
+            # Handle the case where the user has no membership
+            logger.info("User has no membership.")
+
+        if organization:
+            logger.info(organization)
+            return self.__aggregation_response_dynamic(
+                "observable_name", False, users=users_of_organization
+            )
+        else:
+            return self.__aggregation_response_dynamic(
+                "observable_name", False, user=current_user
+            )
+
+    @action(
+        url_path="aggregate/org/md5",
+        detail=False,
+        methods=["GET"],
+    )
+    @cache_action_response(timeout=60 * 5)
+    def aggregate_org_md5(self, request):
+        # this is for file
+        current_user = request.user
+        organization = None
+        users_of_organization = None
+        try:
+            if current_user.has_membership():
+                organization = current_user.membership.organization
+                if organization:
+                    users_of_organization = [
+                        membership.user for membership in organization.members.all()
+                    ]
+
+        except AttributeError:
+            # Handle the case where the user has no membership
+            logger.info("User has no membership.")
+
+        if organization:
+            return self.__aggregation_response_dynamic(
+                "md5", False, users=users_of_organization
+            )
+        else:
+            return self.__aggregation_response_dynamic("md5", False, user=current_user)
+
+    #################################################################
+    def __aggregation_response_static(
+        self, annotations: dict, users=None, user=None
+    ) -> Response:
         delta, basis = self.__parse_range(self.request)
+        filter_kwargs = {"received_request_time__gte": delta}
+        if user:
+            filter_kwargs["user"] = user
+        if users:
+            filter_kwargs["user__in"] = users
```

Comment:
I think you can optimize the code by choosing a single parameter "users" here and change the filter based on the number of items in the list